File: notes-backend/src/main/java/com/gestion/notes_backend/notes/domain/Moyenne.java
package com.gestion.notes_backend.notes.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Moyenne {

    private double valeur;
    private int totalNotes;
    private double coefficientTotal;

    public static Moyenne of(double valeur, int totalNotes, double coefficientTotal) {
        return new Moyenne(valeur, totalNotes, coefficientTotal);
    }

    public String getMention() {
        if (valeur >= 16) return "TRES_BIEN";
        if (valeur >= 14) return "BIEN";
        if (valeur >= 12) return "ASSEZ_BIEN";
        if (valeur >= 10) return "PASSABLE";
        return "INSUFFISANT";
    }

    public boolean isAdmis() {
        return valeur >= 10;
    }
}

File: frontend-react/src/domain/moyenne.ts
export type Mention =
  | "TRES_BIEN"
  | "BIEN"
  | "ASSEZ_BIEN"
  | "PASSABLE"
  | "INSUFFISANT";

export interface Moyenne {
  valeur: number;
  totalNotes: number;
  coefficientTotal: number;
  mention: Mention;
  admis: boolean;
}

const MENTION_LABELS: Record<Mention, string> = {
  TRES_BIEN: "Très bien",
  BIEN: "Bien",
  ASSEZ_BIEN: "Assez bien",
  PASSABLE: "Passable",
  INSUFFISANT: "Insuffisant",
};

export function getMentionLabel(mention: Mention): string {
  return MENTION_LABELS[mention];
}

export function calculerStatut(admis: boolean): "Admis" | "Ajourné" {
  return admis ? "Admis" : "Ajourné";
}
